metagen_tools/tablegen.m

```diff
@@ -0,0 +1,130 @@
+
+function output = tablegen (basepath2)
+    % Get all the videos from the directory
+    file_struct = dir(fullfile(basepath2,'*.avi'));
+    %remove hidden files
+    file_struct = file_struct(~strncmp({file_struct.name}, '.', 1));
+    % Extract base name
+    bstring=char(basepath2);
+    File_name=bstring(1+max(strfind(basepath2,'\')):end); % ***** edit '\' or '/' depending on mac or windows
+    % Extract the timestamps from the filenames
+    timestamps = regexp({file_struct.name}, '-\d+', 'match');
+    timestamps = cellfun(@(x) extractAfter(x,'-'),timestamps,'UniformOutput',false);
+    timestamps = cellfun(@(x) str2double(x),timestamps,'UniformOutput',false);
+    [~, sortedIndices] = sort( [timestamps{:}]);
+    file_struct=file_struct(sortedIndices);
+    subid=[];
```

Comment:
consider a blank csv with headers so you don't have to initialize all variable names here.